### PR TITLE
Fix MaintenanceModeHelper->isActive() for a NULL $matchSessionId

### PR DESCRIPTION
**Pull request by GeorgII-web**
Issue:
In an enabled maintenance mode, the `isActive()` always return `false`. 

Details:
For a  `$this->maintenanceModeHelper->isActive()`  case it was always `false`.
For a  `$this->maintenanceModeHelper->isActive('some_session_id')`  case it worked properly.

Fix:
Now, when `$matchSessionId` is NULL by default and in a DB we have row with `maintanance_mode` it'll return `true` as expected.

  
  

## Changes in this pull request  
Resolves #

### <samp>🤖 Generated by Copilot at e8aecb8</samp>

Allow bypassing maintenance mode for specific sessions. Add a condition in `MaintenanceModeHelper.php` to check if the current session ID matches the maintenance mode entry.

### <samp>🤖 Generated by Copilot at e8aecb8</samp>

> _`maintenance_mode`_
> _Only for other sessions_
> _A winter bypass_

### <samp>🤖 Generated by Copilot at e8aecb8</samp>

*  Add a feature to bypass the maintenance mode for specific sessions ([link](https://github.com/pimcore/pimcore/pull/16235/files?diff=unified&w=0#diff-f16bffa67c7a8d304f9c83fade71a9374e93647a33a12b9324f083dcc506d2c0R58-R60))

### CODE REVIEW

**Comment by blankse** [comment]
@GeorgII-web Target branch should be `11.1` for bug fixes.

**Comment by dvesh3** [comment]
@GeorgII-web please see https://github.com/pimcore/pimcore/pull/16235#issuecomment-1816419781

File: lib/Tool/MaintenanceModeHelper.php [hunk 1]
```diff
@@ -55,6 +55,9 @@ public function deactivate(): void
     public function isActive(string $matchSessionId = null): bool
     {
         if ($this->db->isConnected() && $maintenanceModeEntry = $this->getEntry()) {
+            if ($matchSessionId === null && $maintenanceModeEntry !== null) {
```
Comment (dvesh3) [inline lib/Tool/MaintenanceModeHelper.php:58]:
```suggestion
            if ($matchSessionId === null) {
```
we can ignore `$maintenanceModeEntry` check as it is already true at this point from line above.

**Comment by dvesh3** [comment]
pushed directly to 11.1 branch https://github.com/pimcore/pimcore/commit/29ced9f79af18ef4485255973b1d891026908d73 thanks!

**Pull request closed without merging**